Implemented Gson

TheBusyBiscuit opened the pull request:
Since org.simple.json is already fading out, it is time to use Gson.

I am not sure about some of the implementations that utilised Lists, so correct me on those if you have better suggestions.

rutgerkok commented on src/main/java/nl/rutgerkok/blocklocker/impl/profile/ProfileFactoryImpl.java:
```diff
@@ -175,7 +176,7 @@ public Profile fromRedstone() {
-    private Optional<UUID> getUniqueId(JSONObject object, String key) {
+    private Optional<UUID> getUniqueId(JsonObject object, String key) {
         Object uuidObject = object.get(key);
         if (!(uuidObject instanceof String)) {
```
uuidObject will be a JsonPrimitive, not a String, so this needs to be updated.

rutgerkok commented on src/main/java/nl/rutgerkok/blocklocker/impl/updater/UpdateCheckResult.java:
```diff
@@ -40,8 +41,8 @@
         needsUpdate = getBoolean(object, NEEDS_UPDATE_KEY);
-        latestVersion = Optional.fromNullable((String) object.get(VERSION_KEY));
-        fileMd5 = Optional.fromNullable((String) object.get(MD5_KEY));
+        latestVersion = Optional.fromNullable(object.get(VERSION_KEY).getAsString());
```
If `object.get(VERSION_KEY)` returns null, this will throw a NullPointerException. Same for line below.

TheBusyBiscuit commented:
> I have found two bugs, could you please take a look?

Of course, fixed them accordingly.

rutgerkok commented on src/main/java/nl/rutgerkok/blocklocker/impl/profile/ProfileFactoryImpl.java:
```diff
@@ -177,12 +178,13 @@ public Profile fromRedstone() {
-        if (!(uuidObject instanceof String)) {
+        JsonElement uuidObject = object.get(key);
+        
+        if (!uuidObject.isJsonPrimitive() || !uuidObject.getAsJsonPrimitive().isString()) {
```
This will throw a NullPointerException if uuidObject is null.

TheBusyBiscuit replied:
Right, forgot to replace the null-check that was part of the instanceof...

rutgerkok commented on src/main/java/nl/rutgerkok/blocklocker/impl/updater/UpdateCheckResult.java:
```diff
@@ -41,8 +42,17 @@
+        if (object.has(VERSION_KEY)) {
+        	latestVersion = Optional.fromNullable(object.get(VERSION_KEY).getAsString());
+        }
+        else latestVersion = Optional.absent();
```
Now that your Optional pull request is merged, this can also become `latestVersion = Optional.ofNullable(object.get(VERSION_KEY)).map(JsonElement::getAsString)`. Then we don't need the if statement anymore. But I can fix this after merging if you want, the code works just fine like this after all.

TheBusyBiscuit replied:
Oh yeah, no worries, I will quickly implement that too.

TheBusyBiscuit commented:
Also resolved the merge conflicts.

TheBusyBiscuit commented:
Alright, I removed some unused imports that popped up due to resolving the merge conflicts.
And I also updated the Updater code to use java.util.Optional

rutgerkok commented:
Many thanks, I've merged this!
